vumi_http_retry/workers/sender/worker.py

```diff
@@ -93,19 +97,26 @@ def retry(self, req):
 
     @inlineCallbacks
     def poll(self):
+        limiter = TaskLimiter(
+            self.config.concurrency_limit,
+            errback=self.on_error)
+
         while True:
             if self.stopping:
                 break
 
             req = yield self.next_req()
 
-            if not req:
+            if req is None:
                 break
 
-            yield self.retry(req)
+            yield limiter.add(self.retry, req)
+
+        yield limiter.done()
 
     def on_error(self, err):
         log.err(err)
+        reactor.stop()
```

Comment:
Woops, sorry for unrelated change, I think I was seeing how this affects tests and forgot to remove.